src/msgpack/Converters/BinaryConverter.cs

```diff
@@ -4,7 +4,7 @@ namespace TarantoolDnx.MsgPack.Converters
 {
     internal class BinaryConverter : IMsgPackConverter<byte[]>
     {
-        public void Write(byte[] value, IMsgPackWriter writer, MsgPackContext context)
+        public void Write(byte[] value, IBytesWriter writer, MsgPackContext context)
```

Comment:
BytesReader and BytesWriter are too generic names. JsonWriter from Json.NET library also write bytes, but it called JsonWriter, not BytesWriter.